### DISCUSSION TITLE
In a spec file, are tests (`it`) running asynchronously, serially, or parallelly by default?

### DISCUSSION THREAD

**Discussion by quad5** [cypress-io/cypress#15636] [Questions and Help] [answered]
Are tests (`it`) running asynchronously, serially, or parallelly by default? This is all in one spec file.

 I have 10 tests (for example `test1` through `test10`). These tests are independent to each other as long as I unblock full-screen dialog so next test can have access to its element. And I did the necessary cleanup in each test (`it`), but the test follows it still sees full-page dialog.

What can I do to remove this blockage beside re-arranging test order?

**Comment by herecydev** [comment, marked as answer]
They run serially; so you shouldn't run into issues given what you're describing with the cleanup. Can you post the code or a repro? Is the dialog definitely being cleared by the previous test?

**Comment by quad5** [comment]
Thanks for the reply. Turns out it might be an issue on our website.
